target/linux/octeon/base-files/lib/upgrade/platform.sh

```diff
@@ -83,7 +83,8 @@ platform_check_image() {
 
 	case "$board" in
 	er | \
-	erlite)
+	erlite | \
+	usg3p)
```

Comment:
As you already found out, the device definition will have effect on the sysupgrade-* name used here, and you need BOARD_NAME to alter it if you want to force a certain name for that folder.

A convenient way around that can be found by using a wildcard for the folder name, as done for the generic nand script here:
https://github.com/openwrt/openwrt/blob/master/package/base-files/files/lib/upgrade/nand.sh#L249

This will allow you to drop BOARD_NAME in device definition entirely, and as a bonus upgrade won't break if the board/device name is changed in the future.